Build failure: petsc

### Steps To Reproduce

Steps to reproduce the behavior:
1. build `petsc`

### Build log

These are the last few lines of the build logs.

```
mpicc -fPIC -Wall -Wwrite-strings -Wno-unknown-pragmas -Wno-lto-type-mismatch -Wno-stringop-overflow -fstack-protector -fvisibility=hidden -g3 -O0  -I/build/petsc-3.19.4/include -I/build/petsc-3.19.4/arch-linux-c-debug/include     -Wl,-export-dynamic ex19.c  -Wl,-rpath,/build/petsc-3.19.4/arch-linux-c-debug/lib -L/build/petsc-3.19.4/arch-linux-c-debug/lib -Wl,-rpath,/nix/store/vr8njqmvv9z1wkwrqk7jy56x6lb43l31-openmpi-4.1.6/lib -L/nix/store/vr8njqmvv9z1wkwrqk7jy56x6lb43l31-openmpi-4.1.6/lib -Wl,-rpath,/nix/store/kcyymk901cy4p2x2sqhml7nb9zhab5f8-libfabric-1.21.0-dev/lib -L/nix/store/kcyymk901cy4p2x2sqhml7nb9zhab5f8-libfabric-1.21.0-dev/lib -Wl,-rpath,/nix/store/cl9wrjrc75ir36331icq4i7r4998jiwr-pmix-5.0.1/lib -L/nix/store/cl9wrjrc75ir36331icq4i7r4998jiwr-pmix-5.0.1/lib -Wl,-rpath,/nix/store/vwrz2sj2xxb7jnq38mplqs7fq7dk6x66-libnl-3.8.0-dev/lib -L/nix/store/vwrz2sj2xxb7jnq38mplqs7fq7dk6x66-libnl-3.8.0-dev/lib -Wl,-rpath,/nix/store/hlk3hfg1j6j3rbmbsa8f0g3nnxzvbwvf-libpsm2-12.0.1/lib -L/nix/store/hlk3hfg1j6j3rbmbsa8f0g3nnxzvbwvf-libpsm2-12.0.1/lib -Wl,-rpath,/nix/store/cy6dh661j89s69mxlbi604z06arqb474-blas-3/lib -L/nix/store/cy6dh661j89s69mxlbi604z06arqb474-blas-3/lib -Wl,-rpath,/nix/store/ssjp1ryvwndqf4wa8530zfxpprrp2g5i-lapack-3/lib -L/nix/store/ssjp1ryvwndqf4wa8530zfxpprrp2g5i-lapack-3/lib -Wl,-rpath,/nix/store/35pq4hr29c3sl79lgfwgsvd9nwzyp4am-glibc-2.39-5/lib -L/nix/store/35pq4hr29c3sl79lgfwgsvd9nwzyp4am-glibc-2.39-5/lib -Wl,-rpath,/nix/store/nn152mgpdfcw92b3b03g9kmnlpd9jris-gcc-13.2.0/lib/gcc/x86_64-unknown-linux-gnu/13.2.0 -L/nix/store/nn152mgpdfcw92b3b03g9kmnlpd9jris-gcc-13.2.0/lib/gcc/x86_64-unknown-linux-gnu/13.2.0 -Wl,-rpath,/nix/store/f1ii69v7p27z1r5zybmlbld3bdzm6a5f-gcc-13.2.0-lib/lib -L/nix/store/f1ii69v7p27z1r5zybmlbld3bdzm6a5f-gcc-13.2.0-lib/lib -Wl,-rpath,/nix/store/k8njqq4rc3knarvp24waf7isbvxrpi7d-gfortran-13.2.0-lib/lib -L/nix/store/k8njqq4rc3knarvp24waf7isbvxrpi7d-gfortran-13.2.0-lib/lib -Wl,-rpath,/nix/store/s8i61f6l2xr63kc1wfba5065wwdkayxa-gfortran-wrapper-13.2.0/bin -L/nix/store/s8i61f6l2xr63kc1wfba5065wwdkayxa-gfortran-wrapper-13.2.0/bin -Wl,-rpath,/nix/store/bkhd01jbwzxqjn2qgi95vxhncfrivgga-gfortran-13.2.0/lib/gcc/x86_64-unknown-linux-gnu/13.2.0 -L/nix/store/bkhd01jbwzxqjn2qgi95vxhncfrivgga-gfortran-13.2.0/lib/gcc/x86_64-unknown-linux-gnu/13.2.0 -Wl,-rpath,/nix/store/bkhd01jbwzxqjn2qgi95vxhncfrivgga-gfortran-13.2.0/lib64 -L/nix/store/bkhd01jbwzxqjn2qgi95vxhncfrivgga-gfortran-13.2.0/lib64 -Wl,-rpath,/nix/store/bkhd01jbwzxqjn2qgi95vxhncfrivgga-gfortran-13.2.0/lib -L/nix/store/bkhd01jbwzxqjn2qgi95vxhncfrivgga-gfortran-13.2.0/lib -Wl,-rpath,/nix/store/xyghgs8nf62l24z78qn3273yv82gm0si-petsc-3.19.4/lib -lpetsc -llapack -lblas -lm -lmpi_usempif08 -lmpi_usempi_ignore_tkr -lmpi_mpifh -lmpi -lgfortran -lm -lgcc_s -lquadmath -lstdc++ -lquadmath -o ex19
In file included from /nix/store/ij144ma6vs8acil8r9hgr8xkb1dp9azg-glibc-2.39-5-dev/include/bits/libc-header-start.h:33,
                 from /nix/store/ij144ma6vs8acil8r9hgr8xkb1dp9azg-glibc-2.39-5-dev/include/limits.h:26,
                 from /nix/store/nn152mgpdfcw92b3b03g9kmnlpd9jris-gcc-13.2.0/lib/gcc/x86_64-unknown-linux-gnu/13.2.0/include/limits.h:205,
                 from /nix/store/nn152mgpdfcw92b3b03g9kmnlpd9jris-gcc-13.2.0/lib/gcc/x86_64-unknown-linux-gnu/13.2.0/include/syslimits.h:7,
                 from /nix/store/nn152mgpdfcw92b3b03g9kmnlpd9jris-gcc-13.2.0/lib/gcc/x86_64-unknown-linux-gnu/13.2.0/include/limits.h:34,
                 from /build/petsc-3.19.4/include/petscsystypes.h:16,
                 from /build/petsc-3.19.4/include/petscsys.h:44,
                 from /build/petsc-3.19.4/include/petscvec.h:9,
                 from /build/petsc-3.19.4/include/petscmat.h:7,
                 from /build/petsc-3.19.4/include/petscpc.h:7,
                 from /build/petsc-3.19.4/include/petscksp.h:7,
                 from /build/petsc-3.19.4/include/petscsnes.h:7,
                 from ex19.c:68:
/nix/store/ij144ma6vs8acil8r9hgr8xkb1dp9azg-glibc-2.39-5-dev/include/features.h:414:4: warning: #warning _FORTIFY_SOURCE requires compiling with optimization (-O) [-Wcpp]
  414 | #  warning _FORTIFY_SOURCE requires compiling with optimization (-O)
      |    ^~~~~~~
Possible error running C/C++ src/snes/tutorials/ex19 with 1 MPI process
See https://petsc.org/release/faq/
[1714248497.209873] [localhost:12150:0]       tcp_iface.c:893  UCX  ERROR scandir(/sys/class/net) failed: No such file or directory
[1714248497.212895] [localhost:12150:0]       tcp_iface.c:893  UCX  ERROR scandir(/sys/class/net) failed: No such file or directory
lid velocity = 0.0016, prandtl # = 1., grashof # = 1.
Number of SNES iterations = 2
Possible error running C/C++ src/snes/tutorials/ex19 with 2 MPI processes
See https://petsc.org/release/faq/
[1714248498.625277] [localhost:12179:0]       tcp_iface.c:893  UCX  ERROR scandir(/sys/class/net) failed: No such file or directory
[1714248498.625969] [localhost:12180:0]       tcp_iface.c:893  UCX  ERROR scandir(/sys/class/net) failed: No such file or directory
[1714248498.628325] [localhost:12180:0]       tcp_iface.c:893  UCX  ERROR scandir(/sys/class/net) failed: No such file or directory
[1714248498.628301] [localhost:12179:0]       tcp_iface.c:893  UCX  ERROR scandir(/sys/class/net) failed: No such file or directory
lid velocity = 0.0016, prandtl # = 1., grashof # = 1.
Number of SNES iterations = 2
Possible error running Fortran example src/snes/tutorials/ex5f with 1 MPI process
See https://petsc.org/release/faq/
[1714248502.674702] [localhost:12286:0]       tcp_iface.c:893  UCX  ERROR scandir(/sys/class/net) failed: No such file or directory
[1714248502.676990] [localhost:12286:0]       tcp_iface.c:893  UCX  ERROR scandir(/sys/class/net) failed: No such file or directory
Number of SNES iterations =     3
Completed test examples
Error while running make check
make[1]: *** [makefile:123: check] Error 1
make: *** [GNUmakefile:17: check] Error 2
```

### Additional context

Bisecting shows that the build started to fail after an update to `ucx` (https://github.com/NixOS/nixpkgs/pull/306020)

Looks like something related to `tcp_iface.c`


### Notify maintainers

@cburstedde (listed in `meta.maintainers`)
@twesterhout (last update to the package)

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.26, NixOS, 24.05 (Uakari), 24.05.20240412.cfd6b5f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - channels(root): `""`
 - nixpkgs: `/nix/store/yy5l09gfsagkv8rswblknmsjc2gyr20d-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/libraries/science/math/petsc/filter_mpi_warnings.patch
+++ b/pkgs/development/libraries/science/math/petsc/filter_mpi_warnings.patch
@@ -1,12 +1,12 @@
 diff --git a/src/snes/tutorials/makefile b/src/snes/tutorials/makefile
-index 168febb34b6..71068469066 100644
+index 672a62a..a5fd1c4 100644
 --- a/src/snes/tutorials/makefile
 +++ b/src/snes/tutorials/makefile
 @@ -13,6 +13,7 @@ include ${PETSC_DIR}/lib/petsc/conf/rules
  #  these tests are used by the makefile in PETSC_DIR for basic tests of the install and should not be removed
  testex5f: ex5f.PETSc
  	-@${MPIEXEC} -n 1 ${MPIEXEC_TAIL} ./ex5f -snes_rtol 1e-4 > ex5f_1.tmp 2>&1; \
-+        sed -i '/hwloc\/linux/d ; /ERROR opendir(\/sys\/class\/net) failed/d' ex5f_1.tmp; \
++        sed -i '/hwloc\/linux/d ; /ERROR scandir(\/sys\/class\/net) failed/d' ex5f_1.tmp; \
          if (${DIFF} output/ex5f_1.testout ex5f_1.tmp > /dev/null 2>&1) then \
            echo "Fortran example src/snes/tutorials/ex5f run successfully with 1 MPI process"; \
          else \
@@ -14,7 +14,7 @@ index 168febb34b6..71068469066 100644
          ${MAKE} PETSC_ARCH=${PETSC_ARCH} PETSC_DIR=${PETSC_DIR} ex5f.rm;
  testex19: ex19.PETSc
  	-@${MPIEXEC} -n 1 ${MPIEXEC_TAIL} ./ex19 -da_refine 3 -pc_type mg -ksp_type fgmres  > ex19_1.tmp 2>&1; \
-+        sed -i '/hwloc\/linux/d ; /ERROR opendir(\/sys\/class\/net) failed/d' ex19_1.tmp; \
++        sed -i '/hwloc\/linux/d ; /ERROR scandir(\/sys\/class\/net) failed/d' ex19_1.tmp; \
          if (${DIFF} output/ex19_1.testout ex19_1.tmp > /dev/null 2>&1) then \
            echo "C/C++ example src/snes/tutorials/ex19 run successfully with 1 MPI process"; \
          else \
@@ -22,7 +22,7 @@ index 168febb34b6..71068469066 100644
          ${RM} -f ex19_1.tmp;
  testex19_mpi:
  	-@${MPIEXEC} -n 2 ${MPIEXEC_TAIL} ./ex19 -da_refine 3 -pc_type mg -ksp_type fgmres  > ex19_1.tmp 2>&1; \
-+        sed -i '/hwloc\/linux/d ; /ERROR opendir(\/sys\/class\/net) failed/d' ex19_1.tmp; \
++        sed -i '/hwloc\/linux/d ; /ERROR scandir(\/sys\/class\/net) failed/d' ex19_1.tmp; \
          if (${DIFF} output/ex19_1.testout ex19_1.tmp > /dev/null 2>&1) then \
            echo "C/C++ example src/snes/tutorials/ex19 run successfully with 2 MPI processes"; \
          else \
@@ -30,7 +30,7 @@ index 168febb34b6..71068469066 100644
  #use unpreconditioned norm because HYPRE device installations use different AMG parameters
  runex19_hypre:
  	-@${MPIEXEC} -n 2 ${MPIEXEC_TAIL} ./ex19 -da_refine 3 -snes_monitor_short -ksp_norm_type unpreconditioned -pc_type hypre > ex19_1.tmp 2>&1; \
-+        sed -i '/hwloc\/linux/d ; /ERROR opendir(\/sys\/class\/net) failed/d' ex19_1.tmp; \
++        sed -i '/hwloc\/linux/d ; /ERROR scandir(\/sys\/class\/net) failed/d' ex19_1.tmp; \
            if (${DIFF} output/ex19_hypre.out ex19_1.tmp) then \
              echo "C/C++ example src/snes/tutorials/ex19 run successfully with hypre"; \
            else  \
@@ -38,7 +38,7 @@ index 168febb34b6..71068469066 100644
            ${RM} -f ex19_1.tmp
  runex19_hypre_cuda:
  	-@${MPIEXEC} -n 2 ${MPIEXEC_TAIL} ./ex19 -dm_vec_type cuda -dm_mat_type aijcusparse -da_refine 3 -snes_monitor_short -ksp_norm_type unpreconditioned -pc_type hypre > ex19_1.tmp 2>&1; \
-+        sed -i '/hwloc\/linux/d ; /ERROR opendir(\/sys\/class\/net) failed/d' ex19_1.tmp; \
++        sed -i '/hwloc\/linux/d ; /ERROR scandir(\/sys\/class\/net) failed/d' ex19_1.tmp; \
  	   if (${DIFF} output/ex19_hypre.out ex19_1.tmp) then \
             echo "C/C++ example src/snes/tutorials/ex19 run successfully with hypre/cuda"; \
             else  \
@@ -46,7 +46,7 @@ index 168febb34b6..71068469066 100644
  	   ${RM} -f ex19_1.tmp
  runex19_hypre_hip:
  	-@${MPIEXEC} -n 2 ${MPIEXEC_TAIL} ./ex19 -dm_vec_type hip -da_refine 3 -snes_monitor_short -ksp_norm_type unpreconditioned -pc_type hypre > ex19_1.tmp 2>&1; \
-+        sed -i '/hwloc\/linux/d ; /ERROR opendir(\/sys\/class\/net) failed/d' ex19_1.tmp; \
++        sed -i '/hwloc\/linux/d ; /ERROR scandir(\/sys\/class\/net) failed/d' ex19_1.tmp; \
  	   if (${DIFF} output/ex19_hypre.out ex19_1.tmp) then \
             echo "C/C++ example src/snes/tutorials/ex19 run successfully with hypre/hip"; \
             else \
@@ -54,7 +54,7 @@ index 168febb34b6..71068469066 100644
  	   ${RM} -f ex19_1.tmp
  runex19_cuda:
  	-@${MPIEXEC} -n 1 ${MPIEXEC_TAIL} ./ex19 -snes_monitor -dm_mat_type seqaijcusparse -dm_vec_type seqcuda -pc_type gamg -pc_gamg_esteig_ksp_max_it 10 -ksp_monitor -mg_levels_ksp_max_it 3  > ex19_1.tmp 2>&1; \
-+        sed -i '/hwloc\/linux/d ; /ERROR opendir(\/sys\/class\/net) failed/d' ex19_1.tmp; \
++        sed -i '/hwloc\/linux/d ; /ERROR scandir(\/sys\/class\/net) failed/d' ex19_1.tmp; \
  	   if (${DIFF} output/ex19_cuda_1.out ex19_1.tmp) then \
             echo "C/C++ example src/snes/tutorials/ex19 run successfully with cuda"; \
             else  \
@@ -62,7 +62,7 @@ index 168febb34b6..71068469066 100644
  	   ${RM} -f ex19_1.tmp
  runex19_ml:
  	-@${MPIEXEC} -n 2 ${MPIEXEC_TAIL} ./ex19 -da_refine 3 -snes_monitor_short -pc_type ml > ex19_1.tmp 2>&1; \
-+        sed -i '/hwloc\/linux/d ; /ERROR opendir(\/sys\/class\/net) failed/d' ex19_1.tmp; \
++        sed -i '/hwloc\/linux/d ; /ERROR scandir(\/sys\/class\/net) failed/d' ex19_1.tmp; \
  	   if (${DIFF} output/ex19_ml.out ex19_1.tmp) then  \
             echo "C/C++ example src/snes/tutorials/ex19 run successfully with ml"; \
             else \
@@ -70,7 +70,7 @@ index 168febb34b6..71068469066 100644
             ${RM} -f ex19_1.tmp
  runex19_fieldsplit_mumps:
  	-@${MPIEXEC} -n 2 ${MPIEXEC_TAIL} ./ex19 -pc_type fieldsplit -pc_fieldsplit_block_size 4 -pc_fieldsplit_type SCHUR -pc_fieldsplit_0_fields 0,1,2 -pc_fieldsplit_1_fields 3 -fieldsplit_0_pc_type lu -fieldsplit_1_pc_type lu -snes_monitor_short -ksp_monitor_short  -fieldsplit_0_pc_factor_mat_solver_type mumps -fieldsplit_1_pc_factor_mat_solver_type mumps > ex19_6.tmp 2>&1; \
-+        sed -i '/hwloc\/linux/d ; /ERROR opendir(\/sys\/class\/net) failed/d' ex19_6.tmp; \
++        sed -i '/hwloc\/linux/d ; /ERROR scandir(\/sys\/class\/net) failed/d' ex19_6.tmp; \
  	   if (${DIFF} output/ex19_fieldsplit_5.out ex19_6.tmp) then  \
             echo "C/C++ example src/snes/tutorials/ex19 run successfully with mumps"; \
             else  \
@@ -78,7 +78,7 @@ index 168febb34b6..71068469066 100644
             ${RM} -f ex19_6.tmp
  runex19_superlu_dist:
  	-@${MPIEXEC} -n 1 ${MPIEXEC_TAIL} ./ex19 -da_grid_x 20 -da_grid_y 20 -pc_type lu -pc_factor_mat_solver_type superlu_dist > ex19.tmp 2>&1; \
-+        sed -i '/hwloc\/linux/d ; /ERROR opendir(\/sys\/class\/net) failed/d' ex19.tmp; \
++        sed -i '/hwloc\/linux/d ; /ERROR scandir(\/sys\/class\/net) failed/d' ex19.tmp; \
  	   if (${DIFF} output/ex19_superlu.out ex19.tmp) then \
             echo "C/C++ example src/snes/tutorials/ex19 run successfully with superlu_dist"; \
             else  \
@@ -86,7 +86,7 @@ index 168febb34b6..71068469066 100644
  	   ${RM} -f ex19.tmp
  runex19_suitesparse:
  	-@${MPIEXEC} -n 1 ${MPIEXEC_TAIL} ./ex19 -da_refine 3 -snes_monitor_short -pc_type lu -pc_factor_mat_solver_type umfpack > ex19_1.tmp 2>&1; \
-+        sed -i '/hwloc\/linux/d ; /ERROR opendir(\/sys\/class\/net) failed/d' ex19_1.tmp; \
++        sed -i '/hwloc\/linux/d ; /ERROR scandir(\/sys\/class\/net) failed/d' ex19_1.tmp; \
  	   if (${DIFF} output/ex19_suitesparse.out ex19_1.tmp) then \
             echo "C/C++ example src/snes/tutorials/ex19 run successfully with suitesparse"; \
             else \
@@ -94,7 +94,7 @@ index 168febb34b6..71068469066 100644
  	   ${RM} -f ex19_1.tmp
  runex3k_kokkos: ex3k.PETSc
  	-@OMP_PROC_BIND=false ${MPIEXEC} -n 2 ${MPIEXEC_TAIL} ./ex3k -view_initial -dm_vec_type kokkos -dm_mat_type aijkokkos -use_gpu_aware_mpi 0 -snes_monitor > ex3k_1.tmp 2>&1 ;\
-+        sed -i '/hwloc\/linux/d ; /ERROR opendir(\/sys\/class\/net) failed/d' ex3k_1.tmp; \
++        sed -i '/hwloc\/linux/d ; /ERROR scandir(\/sys\/class\/net) failed/d' ex3k_1.tmp; \
  	if (${DIFF} output/ex3k_1.out ex3k_1.tmp) then \
            echo "C/C++ example src/snes/tutorials/ex3k run successfully with kokkos-kernels"; \
          else \

